awx_collection/plugins/modules/tower_job_template.py

```diff
@@ -12,410 +13,430 @@
+      required: False
+      type: int
+      default: '1'
+    ask_skip_tags_on_launch:
+      required: False
       type: bool
-      default: 'no'
-    ask_credential:
+      default: 'False'
+    name:
       description:
-        - Prompt user for credential on launch.
-      type: bool
-      default: 'no'
-    survey_enabled:
+        - Name of this job template.
+      required: True
+      type: str
+    new_name:
       description:
-        - Enable a survey on the job template.
-      version_added: 2.7
-      type: bool
-      default: 'no'
-    survey_spec:
+        - To use when changing a team's name.
+      required: True
+      type: str
+    use_fact_cache:
       description:
-        - JSON/YAML dict formatted survey definition.
-      version_added: 2.8
+        - If enabled, Tower will act as an Ansible Fact Cache Plugin; persisting facts at the end of a playbook run to the database and caching facts for use by Ansible.
+      required: False
+      type: bool
+      default: 'False'
+    extra_vars:
+      required: False
       type: dict
+      default: ''
+    verbosity:
+      required: False
+      type: str
+      default: '0'
+      choices:
+        - '0'
+        - '1'
+        - '2'
+        - '3'
+        - '4'
+        - '5'
+    ask_credential_on_launch:
       required: False
-    become_enabled:
-      description:
-        - Activate privilege escalation.
       type: bool
-      default: 'no'
-    concurrent_jobs_enabled:
+      default: 'False'
+    project:
+      required: False
+      type: str
+    webhook_service:
       description:
-        - Allow simultaneous runs of the job template.
-      version_added: 2.7
-      type: bool
-      default: 'no'
+        - Service that webhook requests will be accepted from
+      required: False
+      type: str
+      choices:
+        - 'github'
+        - 'gitlab'
     timeout:
       description:
-        - Maximum time in seconds to wait for a job to finish (server-side).
+        - The amount of time (in seconds) to run before the task is canceled.
+      required: False
       type: int
-    custom_virtualenv:
-      version_added: "2.9"
-      description:
-        - Local absolute file path containing a custom Python virtualenv to use.
-      type: str
+      default: '0'
+    ask_scm_branch_on_launch:
       required: False
-      default: ''
+      type: bool
+      default: 'False'
     state:
       description:
         - Desired state of the resource.
-      default: "present"
       choices: ["present", "absent"]
+      default: "present"
       type: str
-
-requirements:
-- ansible-tower-cli >= 3.0.2
-
 extends_documentation_fragment: awx.awx.auth
-
-notes:
-  - JSON for survey_spec can be found in Tower API Documentation. See
-    U(https://docs.ansible.com/ansible-tower/latest/html/towerapi/api_ref.html#/Job_Templates/Job_Templates_job_templates_survey_spec_create)
-    for POST operation payload example.
```

Comment:
the notes still seem valid, although the link doesn't work right.

Reply:
In looking at the old module I don't think the survey spec ever actually did anything. Its variables were not used at all after the arg_spec. Need to do some more research on this but we can also see if we can get the link updated.

Reply:
The JT survey spec did apply the survey spec to a job template.

https://github.com/ansible/awx/blob/7ff5bacce5ccb206ec542a44e03ce982055d0c07/awx_collection/plugins/modules/tower_job_template.py#L310

> Its variables were not used at all after the arg_spec

It didn't need to be, because that line passed all params into the method kwargs by default. It's true that `survey_spec` is not accepted in the API fields, but this is the module using tower-cli, and tower-cli will look for this kwarg and make extra requests in order to apply the survey spec.

And just my 2 cents, outright copying of the params into the request data seems like a good direction to move in. There's no reason t have extra lines of code for every parameter. I'd prefer for this all to just be a dump pipe between Ansible arg processing and the API. There are exceptions in some cases, and survey_spec is one in particular that tower-cli code did extra stuff.